LDMP/jobs/mvc.py

```diff
@@ -265,29 +305,91 @@ def __init__(self, job: models.Job, parent=None):
-        if result == QtWidgets.QMessageBox.Yes:
-            log("About to delete the dataset")
-            for path in self.job.results.local_paths:
-                layers.delete_layer_by_filename(str(path))
-            manager.job_manager.delete_job(self.job)
\ No newline at end of file
+    def __init__(self, job: models.Job, parent=None):
+        super(DatasetDetailsWidget, self).__init__(parent)
+        self.setupUi(self)
+
+        self.job = job
+
+        self.name_le.setText(self.job.params.task_name)
+        self.state_le.setText(self.job.status.value)
+        self.created_at_le.setText(str(self.job.start_date))
+
+        self.load_btn.clicked.connect(self.load_dataset)
+        self.delete_btn.clicked.connect(self.__delete_dataset)
+        self.open_directory_btn.clicked.connect(self.open_job_directory)
+        self.export_btn.clicked.connect(self.export_dataset)
+        self.delete_btn.setEnabled(
+            self.job.status == models.JobStatus.DOWNLOADED
+        )
+
+        self.alg_le.setText(self.job.script.name)
+        self.path = None
+
+        for path in self.job.results.local_paths:
+            if 'tif' in str(path):
+                self.path = str(path)
+        self.path_le.setText(self.path)
+
+        self.load_btn.setIcon(
+            QtGui.QIcon(':/plugins/LDMP/icons/mActionAddRasterLayer.svg'))
+        self.open_directory_btn.setIcon(
+            QtGui.QIcon(':/images/themes/default/mActionFileOpen.svg'))
+        self.export_btn.setIcon(
+            QtGui.QIcon(':/plugins/LDMP/icons/export_zip.svg'))
+        self.delete_btn.setIcon(
+            QtGui.QIcon(':/plugins/LDMP/icons/mActionDeleteSelected.svg'))
+
+        self.load_job_details()
+
+        self.bar = qgis.gui.QgsMessageBar()
+        self.bar.setSizePolicy(
+            QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Fixed
+        )
+        self.layout().addWidget(self.bar, 0, 0, alignment=QtCore.Qt.AlignTop)
+
+    def __delete_dataset(self):
+        self.delete_dataset()
+        self.close()
+
+    def export_dataset(self):
+        log(f"Exporting dataset {self.job.params.task_name!r}")
+        self.export_btn.setEnabled(False)
+
+        # collect all files related to the dataset and compress them
+        # into one zip file.
+        dataset_base_dir = Path(os.path.dirname(self.path))
+        file_name = os.path.splitext(os.path.basename(self.path))[0]
+        folder_contents = dataset_base_dir.glob(f"{file_name}.*")
+        files = [f for f in folder_contents if f.is_file()]
+
+        manager.job_manager.exports_dir.mkdir(exist_ok=True)
+
+        try:
+            zipped_file = f"{str(manager.job_manager.exports_dir)}/" \
+                          f"{self.job.params.task_name}.zip"
+            with ZipFile(zipped_file, 'w') as zip:
+                for file in files:
+                    zip.write(file, os.path.basename(file))
+        except Exception:
```

Comment:
Always avoid using the generic `Exception` in a try block. Check which exceptions can happen and write code to guard against those.

If we ever get some unforeseen exception we fix the code